### PR TITLE
refactor(rn tester app): change dimensions example to hooks

### PR DESCRIPTION
**Pull request by Marcoo09**
## Summary
This pull request migrates the dimensions example to using React Hooks.

## Changelog
[General] [Changed] - RNTester: Migrate Dimensions to hooks

## Test Plan
The animation works exactly as it did as when it was a class component

### CODE REVIEW

**Review by NickGerleman — CHANGES_REQUESTED** [review]
Thanks for the contribution! Please take a look at the flow errors surfaced in the `test_js` CircleCI job. You should also be able to run this locally via `yarn flow-check-android` and `yarn flow-check-ios`.

File: packages/rn-tester/js/examples/Dimensions/DimensionsExample.js [hunk 1]
```diff
@@ -8,38 +8,28 @@
...
     );
-  }
 
-  componentWillUnmount() {
-    this._dimensionsSubscription?.remove();
-  }
+    return () => {
+      dimensionsSubscription?.current?.remove();
+    };
+  }, [props.dim]);
```
Comment (hoxyq) [inline packages/rn-tester/js/examples/Dimensions/DimensionsExample.js:30]: I think you can make this work without using `dimensionsSubscription` variable outside of the `useEffect` hook

Just initialize variable with a subscription inside the hook

Comment (Marcoo09) [reply packages/rn-tester/js/examples/Dimensions/DimensionsExample.js:30]: Thank, I've just changed it

**Review by NickGerleman — APPROVED** [review]
LGTM

**Pull request closed without merging**